example/spec/parallel/testAPI/granuleSpec.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const { s3PutObject } = require('@cumulus/aws-client/S3');
+const { createCollection } = require('@cumulus/integration-tests/Collections');
+const { constructCollectionId } = require('@cumulus/message/Collections');
+const {
+  buildRandomizedGranule,
+} = require('@cumulus/integration-tests/Granules');
+const { deleteCollection } = require('@cumulus/api-client/collections');
+const {
+  createGranule,
+  deleteGranule,
+  getGranule,
+} = require('@cumulus/api-client/granules');
+const { randomId } = require('@cumulus/common/test-utils');
+const { loadConfig } = require('../../helpers/testUtils');
+
+describe('The Granules API', () => {
+  let beforeAllFailed = false;
+  let config;
+  let collection;
+  let collectionId;
+  let granuleId;
+  let prefix;
+  let granuleFile;
+  let randomGranuleRecord;
+
+  beforeAll(async () => {
+    try {
+      config = await loadConfig();
+      prefix = config.stackName;
+
+      collection = await createCollection(prefix);
+      collectionId = constructCollectionId(collection.name, collection.version);
+
+      granuleFile = {
+        bucket: config.buckets.public.name,
+        key: randomId('key'),
+      };
+      await s3PutObject({
+        Bucket: granuleFile.bucket,
+        Key: granuleFile.key,
+        Body: 'testfile',
+      });
+
+      randomGranuleRecord = buildRandomizedGranule({
+        collectionId,
+        files: [granuleFile],
+      });
+      granuleId = randomGranuleRecord.granuleId;
+    } catch (error) {
+      beforeAllFailed = true;
+      console.log(error);
+    }
+  });
+
+  afterAll(async () => {
+    await deleteGranule({ prefix, granuleId });
+    await deleteCollection({
+      prefix,
+      collectionName: collection.name,
+      collectionVersion: collection.version,
+    });
+  });
+
+  describe('the Granule Api', () => {
+    it('creates a granule.', async () => {
+      if (beforeAllFailed) {
+        fail('beforeAll() failed');
+      } else {
+        const response = await createGranule({
+          prefix,
+          body: randomGranuleRecord,
+        });
+
+        expect(response.statusCode).toBe(200);
+        const { result } = JSON.parse(response.body);
+        expect(result).toBe(`Wrote Granule ${granuleId}`);
+      }
+    });
+
+    it('can discover the granule in the API.', async () => {
+      const granule = await getGranule({
+        prefix,
+        granuleId,
+      });
+      expect(granule.files[0].size).toEqual(jasmine.any(Number));
+      delete granule.files[0].size;
+      granule.execution = undefined;
```

Comment:
Why are these two lines necessary?

Reply:
L88 because I don't know the size of the file when I create it, oh I guess I do because the body is just 'testbody' but size it isn't part of the input granule. But I would either have to add a size key to the randomGranuleRecord, or delete it from the returned value.
L89 I didn't want to set up an execution, so created a granule without one.  But it gets stripped out of the return value, so I set it to undefined explicitly because it's in the test randomGranuleRecord. 
Both are just to use the expectation on the next line

Reply:
For this first issue, can we just add a `size` key to `granuleFile` around line 44 since we do know the size of the body?

For the second issue, can we remove `  execution: undefined,` from `randomGranuleRecord`? Omitting the property in the return from `randomGranuleRecord` would have the same effect of returing a granule that has no execution, but there wouldn't be an explicit `execution` property set that we have to compare against

Reply:
Yup, Had done (most of) that in the update granule PR,  Can pull it in here. 5eb6f6836d